src/ucp/core/ucp_request.inl

```diff
@@ -164,6 +164,28 @@ void ucp_request_recv_generic_dt_finish(ucp_request_t *req)
     }
 }
 
+static UCS_F_ALWAYS_INLINE void
+ucp_request_send_state_set(ucp_request_t *req, size_t data_offset,
+                           int comp_count, unsigned proto)
+{
+    req->send.state.offset = data_offset;
+    ucs_assert(req->send.state.offset <= req->send.length);
+
+    switch (proto) {
```

Comment:
i see this function is not really doing anything in this PR (BCOPY is a no-op), is this a preparation for next PR? if so, can this be moved to the next PR?

Reply:
This function initializes `req->send.state.offset = data_offset` couple lines above. This is common code for all protocols, `uct_comp` is not used in BCOPY. ZCOPY will be added in next PR.

Reply:
so why need to introduce that function in this PR? can just do `req->send.state.offset = data_offset` in the single place it is used..

Reply:
This is a part of big refactoring with code re-usage and unification for tag and stream APIs. If we will move all changes to that PR, it will be more 500 lines.